plugin/src/main/java/com/iridium/iridiumcore/utils/SkinUtils.java

```diff
@@ -15,25 +15,43 @@
 public class SkinUtils {
 
     private static final HashMap<UUID, String> cache = new HashMap<>();
+    private static final HashMap<String, UUID> uuidCache = new HashMap<>();
     private static final Gson gson = new Gson();
 
+    public static UUID getUUID(String username) {
+        try {
+            if (uuidCache.containsKey(username.toLowerCase())) return uuidCache.get(username.toLowerCase());
+
+            String signature = getURLContent(
+                    "https://api.mojang.com/users/profiles/minecraft/" + username);
+            JsonObject profileJsonObject = gson.fromJson(signature, JsonObject.class);
+            String value = profileJsonObject.get("id").getAsString();
+            if (value == null) return null;
+            return uuidCache.put(username.toLowerCase(),
+                    UUID.fromString(value.replaceFirst(
+                            "(\\p{XDigit}{8})(\\p{XDigit}{4})(\\p{XDigit}{4})(\\p{XDigit}{4})(\\p{XDigit}+)",
```

Comment:
I don't even know how that Regex is supposed to work but I don't think it does. Also pretty over-complicated
```suggestion
                            "(.{8})(.{4})(.{4})(.{4})(.{12})",
```

If you want - to be ignored use
```suggestion
                            "([^-]{8})([^-]{4})([^-]{4})([^-]{4})([^-]{12})",
```

Reply:
Your regex supports invalid uuids. Not that it matters in this context.

Reply:
Sure, but it doesn't matter.

Replace the dots with `[a-f0-9]` so it's also validating if you want that

Reply:
Your regex does the same but is not well readable imo